### ISSUE TITLE
[feature request] torch.onnx.export dynamic_axes to support strings in addition to existing dicts

### ISSUE THREAD

**Issue by vadimkantorov** [pytorch/pytorch#32637]
`dynamic_axes = dict(x = {0 : 'B', 1 : 'T'}, logits = {0 : 'B', 1 : 'C', 2: 't'})` is a bit more verbose than ` dynamic_axes = dict(x = 'BT', logits = 'BCt')`

cc @houseroad @spandantiwari @lara-hdr @BowenBao @neginraoof

**Comment by garymm** [comment]
Yes this would be less verbose, but it would only support the case where all dynamic axes start at 0 and there are no gaps. So we'd have to support both, and I don't think the benefits of the less verbosity for one case is worth the added complexity of having multiple ways to do the same thing.
